lms/djangoapps/ccx/overrides.py

```diff
@@ -142,18 +142,15 @@ def override_field_for_ccx(ccx, block, name, value):
         override_has_changes = serialized_value != override.value
 
     if not override:
-        try:
-            override = CcxFieldOverride.objects.create(
-                ccx=ccx,
-                location=block.location,
-                field=name,
-                value=serialized_value
-            )
+        override, created = CcxFieldOverride.objects.get_or_create(
+            ccx=ccx,
+            location=block.location,
+            field=name,
+            defaults = {'value': serialized_value},
+        )
+        if created:
             _get_overrides_for_ccx(ccx).setdefault(block.location, {})[name + "_id"] = override.id
-        except IntegrityError:
-            transaction.commit()
-            kwargs = {'ccx': ccx, 'location': block.location, 'field': name}
-            override = CcxFieldOverride.objects.get(**kwargs)
+        else:
```

Comment:
@adampalay Yes, to an extent. But this function is only called from 3 Instructor dashboard CCX tab views so I doubt we will be seeing many parallel requests causing `IntegrityErrors`. If we do we can refactor the views a bit so that we can put `outer_atomic` around this whole function or maybe shift to `READ COMMITTED` for the views which do create `IntegrityErrors`.

I have put in a ticket for this: https://openedx.atlassian.net/browse/TNL-3614